pkg/polaris/graphql/aws/exocompute.go

```diff
@@ -111,10 +116,29 @@ type ExocomputeConfigCreate struct {
 func (a API) CreateExocomputeConfig(ctx context.Context, id uuid.UUID, config ExocomputeConfigCreate) (ExocomputeConfig, error) {
 	a.GQL.Log().Print(log.Trace)
 
-	buf, err := a.GQL.Request(ctx, createAwsExocomputeConfigsQuery, struct {
+	query := createAwsExocomputeConfigsQuery
+	var vars interface{} = struct {
 		ID      uuid.UUID                `json:"cloudAccountId"`
 		Configs []ExocomputeConfigCreate `json:"configs"`
-	}{ID: id, Configs: []ExocomputeConfigCreate{config}})
+	}{ID: id, Configs: []ExocomputeConfigCreate{config}}
+
+	if graphql.VersionOlderThan(a.Version, "master-52101", "v20221116") {
+		query = createAwsExocomputeConfigsV0Query
+		type legacyExocomputeConfigCreate struct {
+			Region                 Region   `json:"region"`
+			VPCID                  string   `json:"vpcId"`
+			Subnets                []Subnet `json:"subnets"`
+			IsManagedByRubrik      bool     `json:"isPolarisManaged"`
+			ClusterSecurityGroupId string   `json:"clusterSecurityGroupId"`
+			NodeSecurityGroupId    string   `json:"nodeSecurityGroupId"`
+		}
+		vars = struct {
+			ID      uuid.UUID                      `json:"cloudAccountId"`
+			Configs []legacyExocomputeConfigCreate `json:"configs"`
+		}{ID: id, Configs: []legacyExocomputeConfigCreate{(legacyExocomputeConfigCreate)(config)}}
+
+	}
+	buf, err := a.GQL.Request(ctx, query, vars)
```

Comment:
It's not an error to provide GraphQL variables that aren't used by the query. So I think we could write this as:
```
query := createAwsExocomputeConfigsQuery
if graphql.VersionOlderThan(a.Version, "master-52101", "v20221116") {
	query = createAwsExocomputeConfigsV0Query
}
buf, err := a.GQL.Request(ctx, query, struct {
	ID      uuid.UUID     `json:"cloudAccountId"`
	Configs []interface{} `json:"configs"`
}{
	ID: id,
	Configs: []interface{}{struct {
		Region                 Region   `json:"region"`
		VPCID                  string   `json:"vpcId"`
		Subnets                []Subnet `json:"subnets"`
		IsPolarisManaged       bool     `json:"<old-name>"`
		IsManagedByRubrik      bool     `json:"<new-name>"`
		ClusterSecurityGroupId string   `json:"clusterSecurityGroupId"`
		NodeSecurityGroupId    string   `json:"nodeSecurityGroupId"`
	}{Region: config.Region, VPCID: config.VPCID, Subnets: config.Subnets, IsPolarisManaged: config.IsManagedByRubrik, IsManagedByRubrik: config.IsManagedByRubrik, ClusterSecurityGroupId: config.ClusterSecurityGroupId, NodeSecurityGroupId: config.NodeSecurityGroupId}},
})
if err != nil {
	return ExocomputeConfig{}, fmt.Errorf("failed to request CreateExocomputeConfig: %v", err)
}

a.GQL.Log().Printf(log.Debug, "%s(%q, %v): %s", graphql.QueryName(query), id, config, string(buf))
```
Which looks a bit more like the rest of the requests.

Reply:
As per our Slack conversation it seems like it is an error in this case, leaving as is.